pkg/authentication/bootstrapauthenticator/bootstrap.go

```diff
@@ -151,3 +132,37 @@ func (b *bootstrapUserDataGetter) Get() (*BootstrapUserData, bool, error) {
 		UID:          base64.RawURLEncoding.EncodeToString(uidBytes[:]),
 	}, true, nil
 }
+
+func (b *bootstrapUserDataGetter) IsEnabled() (bool, error) {
+	secret, err := b.getBootstrapUserSecret()
+	if err == errSecretRecreated {
+		return false, nil
+	}
+	if err != nil || secret == nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (b *bootstrapUserDataGetter) getBootstrapUserSecret() (*corev1.Secret, error) {
+	secret, err := b.secrets.Get(bootstrapUserBasicAuth, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		klog.V(4).Infof("%s secret does not exist", bootstrapUserBasicAuth)
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	if secret.DeletionTimestamp != nil {
+		klog.V(4).Infof("%s secret is being deleted", bootstrapUserBasicAuth)
+		return nil, nil
+	}
+	namespace, err := b.namespaces.Get(metav1.NamespaceSystem, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if secret.CreationTimestamp.After(namespace.CreationTimestamp.Add(time.Hour)) {
```

Comment:
what is this heuristic? It needs docs why we do this.

Reply:
this is supposed to make it impossible for the bootstrap user to be configured again after it was once removed